### PR TITLE
Use optparse-applicative for cmd-line arguments

### LINKED ISSUE #17: Usage and help
Write one in the readme and have help argument in the CLI.

### PR DESCRIPTION
**Pull request by oleks**
Thereby adding usage and help.

Closes #17

### CODE REVIEW

File: app/Main.hs [hunk 1]
```diff
@@ -244,52 +246,127 @@ findDelimiter (s:ss) =
...
+  deriving (Show)
+
+cmdFeedbackParser :: App.Parser Command
+cmdFeedbackParser =
+  CmdFeedback
+    <$> switch
+      ( long "with-points" )
+  <**> helper
+
+cmdWithDepthParser :: Read a => (Maybe a -> Command) ->  App.Parser Command
```
Comment (oleks) [inline app/Main.hs:279]: I would rather use a `Word` or even `Word8` explicitly here, but the command implementations use `Int` for some reason, so let's fix that in a follow-up PR.

Comment (oleks) [reply app/Main.hs:279]: > I would rather use a `Word` or even `Word8` explicitly here, but the command implementations use `Int` for some reason, so let's fix that in a follow-up PR.

That being the `pending` and `status` commands.

**Comment by oleks** [comment]
The goal of this PR is to introduce `optparse-applicative`, not necessarily clean up the command-line UI.

File: app/Main.hs [hunk 2]
```diff
@@ -244,52 +246,127 @@ findDelimiter (s:ss) =
...
+   <> command "valid" (info (pure CmdValid) (progDesc
+        "validate structure of files at PATHs"))
+   <> command "check" (info (pure CmdCheck) (progDesc
+        "validate and check that points have been given at PATHs"))
+   <> command "show" (info (pure CmdShow) (progDesc
+        "validate, check, and pretty-print"))
+   <> command "pending" (info (cmdWithDepthParser CmdPending) (progDesc
+        "show judgements not yet marked"))
+   <> command "status" (info (cmdWithDepthParser CmdStatus) (progDesc
+        "show status(?)"))
```
Comment (oleks) [inline app/Main.hs:309]: There are too many conflating terms here. "valid", "check", "status". I think just one should really suffice, or some clear explanation as to the difference is in order. Again, this might be subject matter for another PR.

File: app/Main.hs (hunk 1 above)
Comment (athas) [reply app/Main.hs:279]: Word? Is the depth a bitfield?

Comment (oleks) [reply app/Main.hs:279]: > Word? Is the depth a bitfield?

It is a non-negative integer.

Comment (athas) [reply app/Main.hs:279]: Don't use unsigned integers for anything but bitfields. If you think there is a significant source of bugs to be solved by clarifying the type here, then use something like `Nat`, but consider whether you're just adding a bunch of noisy `fromIntegral` calls to the code, or whether you're really fixing a class of bugs.

File: app/Main.hs (hunk 2 above)
Comment (athas) [reply app/Main.hs:309]: Maybe the solution is to group the commands into those you use frequently while working (`status`) and those you only use when you are setting up a marking structure initially (`valid`).

**Review by athas — APPROVED** [review]
I normally have a preference for old-school getopt, but this option parser is pretty concise.
